pixeltable/utils/parquet.py

```diff
@@ -0,0 +1,132 @@
+import json
+import PIL.Image
+import io
+import pyarrow.parquet as pq
+import pyarrow as pa
+import numpy as np
+from pathlib import Path
+from collections import deque
+
+from pixeltable.dataframe import DataFrame
+from typing import List, Tuple, Any, Dict
+
+from pixeltable.utils.transactional_folder import transactional_folder
+
+
+def _write_batch(value_batch : Dict[str, deque], schema : pa.Schema, output_path : Path) -> None:
+    pydict = {}
+    for field in schema:
+        if isinstance(field.type, pa.FixedShapeTensorType):
+            stacked_arr = np.stack(value_batch[field.name])
+            pydict[field.name] = pa.FixedShapeTensorArray.from_numpy_ndarray(stacked_arr)
+        else:
+            pydict[field.name] = value_batch[field.name]
+
+    tab = pa.Table.from_pydict(pydict, schema=schema)
+    pq.write_table(tab, output_path)
+
+
+def save_parquet(df : DataFrame, dest_path : Path, partition_size_bytes : int = 100_000_000):
+    """
+        Internal method to stream dataframe data to parquet format.
+        Does not materialize the dataset to memory.
+
+        It preserves pixeltable type metadata in a json file, which would otherwise 
+        not be available in the parquet format.
+
+        Images are stored inline in a compressed format in their parquet file.
+
+        NB. pixeltable types
+            - string ok
+            - image ok (inlined)
+            - video ok (path only)
```

Comment:
what happens with the other data types? at least int/float/bool should work, no?